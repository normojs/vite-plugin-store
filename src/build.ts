import { buildSync, transformSync } from 'esbuild'
export const buildModule = (fullPath: string, isTs = true) => {
  const result = buildSync({
    entryPoints: [fullPath],
    outfile: 'out.js',
    write: false,
    platform: 'browser',
    bundle: false,
    metafile: true,
    format: isTs ? 'esm' : 'cjs',
  })
  const { text } = result.outputFiles[0]
  return { text, metafile: result.metafile }
}

export const transformModule = (text: string) => {
  return transformSync(text)
}
