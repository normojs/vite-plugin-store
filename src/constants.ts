export const MODULE_IDS = ['normo:store', 'virtual:generated-store', 'store-generated']
export const MODULE_ID_VIRTUAL = '/@vite-plugin-store/generated-store'

/*
@see  https://next.vuex.vuejs.org/zh/guide/hot-reload.html
      > Vuex 支持在开发过程中热重载 mutation、module、action 和 getter
*/
export const HMR_MODULE_NAMES = ['mutations', 'actions', 'getters']
