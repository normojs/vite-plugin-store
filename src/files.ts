import fg from 'fast-glob'
import { ResolvedOptions } from './types'
import { extensionsToGlob } from './utils'

/**
 * Resolves the files that are valid pages for the given context.
 */
export async function getFilesFromPath(path: string, options: ResolvedOptions): Promise<string[]> {
  const {
    extensions,
    exclude,
  } = options

  const ext = extensionsToGlob(extensions)

  const files = await fg(`**/*.${ext}`, {
    ignore: ['node_modules', '.git', '**/__*__/*', ...exclude],
    onlyFiles: true,
    cwd: path,
  })

  return files
}
