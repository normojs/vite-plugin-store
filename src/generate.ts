/**
 * https://github.com/brattonross/vite-plugin-voie/blob/main/packages/vite-plugin-voie/src/routes.ts
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file at
 * https://github.com/brattonross/vite-plugin-voie/blob/main/LICENSE
 */

import fs from 'fs'
import { join } from 'path'
import deepEqual from 'deep-equal'
import { Route, ResolvedOptions, PageDirOptions } from './types'
import { debug, isDynamicRoute, isCatchAllRoute, normalizePath, findRouteByFilename } from './utils'
import { stringifyRoutes } from './stringify'
import { parseCustomBlock, parseSFC } from './parseSfc'

function prepareRoutes(
  routes: Route[],
  options: ResolvedOptions,
  pagesDirOptions: PageDirOptions,
  parent?: Route,
) {
  for (const route of routes) {
    if (route.name) {
      route.name = route.name.replace(/-index$/, '')
      if (pagesDirOptions.baseRoute)
        route.name = `${pagesDirOptions.baseRoute}-${route.name}`
    }

    if (parent) {
      route.path = route.path.replace(/^\//, '')
    }
    else {
      if (pagesDirOptions.baseRoute) {
        const baseRoute = `/${pagesDirOptions.baseRoute}`
        route.path = route.path === '/'
          ? baseRoute
          : baseRoute + route.path
      }
    }

    route.props = true

    if (route.children) {
      delete route.name
      route.children = prepareRoutes(route.children, options, pagesDirOptions, route)
    }
    const filePath = normalizePath(join(options.root, route.component))
    const content = fs.readFileSync(filePath, 'utf8')
    const parsed = parseSFC(content)
    const routeBlock = parsed.customBlocks.find(b => b.type === 'route')

    if (routeBlock)
      Object.assign(route, parseCustomBlock(routeBlock, filePath, options))

    if (typeof options.extendRoute === 'function')
      Object.assign(route, options.extendRoute(route, parent) || {})
  }
  return routes
}

export function generateRoutes(filesPath: string[], pagesDirOptions: PageDirOptions, options: ResolvedOptions): Route[] {
  const { dir: pagesDir } = pagesDirOptions
  const {
    nuxtStyle,
    extensionsRE,
  } = options

  const routes: Route[] = []

  for (const filePath of filesPath) {
    const resolvedPath = filePath.replace(extensionsRE, '')
    const pathNodes = resolvedPath.split('/')

    const component = `/${pagesDir}/${filePath}`
    const route: Route = {
      name: '',
      path: '',
      component,
    }

    let parentRoutes = routes

    for (let i = 0; i < pathNodes.length; i++) {
      const node = pathNodes[i]
      const isDynamic = isDynamicRoute(node, nuxtStyle)
      const isCatchAll = isCatchAllRoute(node, nuxtStyle)
      const normalizedPart = (
        isDynamic
          ? nuxtStyle
            ? isCatchAll ? 'all' : node.replace(/^_/, '')
            : node.replace(/^\[(\.{3})?/, '').replace(/\]$/, '')
          : node
      ).toLowerCase()

      route.name += route.name ? `-${normalizedPart}` : normalizedPart

      // Check nested route
      const parent = parentRoutes.find(node => node.name === route.name)

      if (parent) {
        parent.children = parent.children || []
        parentRoutes = parent.children
        route.path = ''
      }
      else if (normalizedPart === 'index' && !route.path) {
        route.path += '/'
      }
      else if (normalizedPart !== 'index') {
        if (isDynamic) {
          route.path += `/:${normalizedPart}`
          // Catch-all route
          if (isCatchAll)
            route.path += '(.*)'
        }
        else {
          route.path += `/${normalizedPart}`
        }
      }
    }

    parentRoutes.push(route)
  }

  const preparedRoutes = prepareRoutes(routes, options, pagesDirOptions)

  return preparedRoutes
}

export function generateClientCode(routes: Route[], options: ResolvedOptions) {
  const { imports, stringRoutes } = stringifyRoutes(routes, options)

  return `${imports.join('\n')}\n\nconst routes = ${stringRoutes}\n\nexport default routes`
}

export function updateRouteFromHMR(content: string, filename: string, routes: Route[], options: ResolvedOptions): boolean {
  const parsed = parseSFC(content)
  const routeBlock = parsed.customBlocks.find(b => b.type === 'route')
  if (routeBlock) {
    const route = findRouteByFilename(routes, filename)

    if (route) {
      const before = Object.assign({}, route)
      const customBlockContent = parseCustomBlock(routeBlock, filename, options)
      debug.hmr('custom block: %O', customBlockContent)
      Object.assign(route, customBlockContent)
      return !deepEqual(before, route)
    }
  }
  return false
}
