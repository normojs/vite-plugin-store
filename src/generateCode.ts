export const generateCode = function(root: any) {
  return `
  
  import { createStore } from 'vuex'
  // ================= 生成了store code ==================
  ${root.code}
  export const root = ${JSON.stringify({ ...root, code: '' })}
  // ================= end store code ==================

  /* 热加载 */

  // ================= 公共方法 ==================
  /**
   * 设置generatedStore下modules属性namespaced
   * @params {Object} 生成的store code，即：createStore(store)中的store
   */
  function setModulesNamespaced(moduleRoot) {
    if (!moduleRoot.modules) {
      // 结束回调
      return true
    }
    for (const moduleKey in moduleRoot.modules) {
      const module = moduleRoot.modules[moduleKey]
      if (module.namespaced === undefined) {
        // namespaced 默认值为false|undefined
        module.namespaced = true
      }
      setModulesNamespaced(module)
    }// end for
  }
  // ================= end 公共方法 ==================

  // 判断是否开启全局 namespaced，默认开启全局命名空间
  if (generatedStore.namespaced === true) {
    // 如果全局开启，则添加到模块
    setModulesNamespaced(generatedStore)
  }
  const _store = window.store
  if (!window.store) {
    console.log('不存在store, 创建generatedStore： ', generatedStore)
    window.store = createStore(generatedStore)
  }
  export const store = window.store
  export { generatedStore }

  const loadingTime = Date.now()
  const lastHotTime = loadingTime
  console.log('[vite-plugin-store] loading: ', loadingTime)

  let globalVar = 0

  let hotEvent = null
  if (import.meta.hot) {
    import.meta.hot.on('vite-plugin-store-update', (data) => {
      // TODO: 有时会有两次的更新，做防止抖动
      globalVar++
      // 判断类型
      hotEvent = data
      console.log('通知：自定义---------------------------globalVar', globalVar, loadingTime, data)
    })

    import.meta.hot.accept((newModule) => {
      console.log('操作：accept state update ...hotEvent', hotEvent, loadingTime)
      if (hotEvent) {
        const newStore = newModule.generatedStore
        console.log('newStore ', newStore)
        if (hotEvent.type === 'state') {
          // 设置state
          // TODO: 获取state，然后设置state

          let hotState = null
          // 判断是不是/index.ts
          if (hotEvent.moduleInType === hotEvent.moduleName === 'index') {
            //
            hotState = newStore.state()
          }
          else {
            // moduleName: "user/role/element"
            const subModules = hotEvent.moduleName.split('/')
            // TODO:
          }

          // TODO: 循环更新state： state => state.[user.role.menu].info,

          window.store.hotUpdate(newStore)
        }
        else if (hotEvent.type === 'hot-update') {
          window.store.hotUpdate(newStore)
        }
      }

      // console.log('newStore.modules: ', newModule, newStore)
      // let stateData = newStore.modules.account.state()

      // window.store.commit('account/__init__', stateData)

      // TODO: window.store.commit('account/__init__', stateData)
      /*
        TODO: 在handleHotUpdate方法中自动执行

        判断修改的文件路径，如果是state、index文件
        判断是否更改了state，比较出差异
        如果更改了state，则循环被修改的

        // state.user.role.menu.info
        let diffObj = {...} // 如： {'user/role/menu': {path: 'new path'}}
        window.store.commit('__hot_update_state__', diffObj)

      */
    })// end hot.accept
  }// end if

  // ============================================================

  `
}
