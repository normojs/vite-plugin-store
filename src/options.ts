import { UserOptions, ResolvedOptions, PageDirOptions } from './types'

export function resolveOptions(userOptions: UserOptions): ResolvedOptions {
  const {
    storeDir = 'src/store',
    pagesDir = ['src/pages'],
    extensions = ['ts', 'js'],
    importMode = 'async',
    routeBlockLang = 'json5',
    exclude = [],
    syncIndex = true,
    replaceSquareBrackets = false,
    nuxtStyle = false,
  } = userOptions

  const root = process.cwd()

  let pagesDirOptions: PageDirOptions[] = []

  if (typeof pagesDir === 'string') {
    pagesDirOptions = pagesDirOptions.concat({ dir: pagesDir, baseRoute: '' })
  }
  else {
    for (const dir of pagesDir) {
      if (typeof dir === 'string')
        pagesDirOptions = pagesDirOptions.concat({ dir, baseRoute: '' })
      else if (dir as PageDirOptions)
        pagesDirOptions = pagesDirOptions.concat(dir)
    }
  }

  const extensionsRE = new RegExp(`\\.(${extensions.join('|')})$`)

  return Object.assign(
    {},
    {
      routeBlockLang,
      root,
      storeDir,
      pagesDir,
      pagesDirOptions,
      extensions,
      importMode,
      exclude,
      syncIndex,
      replaceSquareBrackets,
      nuxtStyle,
      extensionsRE,
    },
    userOptions,
  )
}
