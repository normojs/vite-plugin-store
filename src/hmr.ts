
import { ViteDevServer } from 'vite'
import { debug, normalizePath, slash, getPathName } from './utils'
import { ResolvedOptions } from './types'
import { HMR_MODULE_NAMES } from './constants'
export function handleHMR(
  server: ViteDevServer,
  generateRoot: any,
  options: ResolvedOptions,
  storeFilePaths: string[],
) {
  const { ws, watcher } = server

  function fullReload() {
    // invalidate module
    // getPagesVirtualModule(server)
    // clearRoutes()
    ws.send({
      type: 'full-reload',
    })
  }

  watcher.on('add', (file) => {
    const path = slash(file)
    // if (isTarget(path, options)) {
    // addPage(pages, path, options)
    debug.hmr('add', path, storeFilePaths)
    fullReload()
    // }
  })
  watcher.on('unlink', (file) => {
    const path = slash(file)
    // if (isTarget(path, options)) {
    // removePage(pages, path)
    debug.hmr('remove', path)
    fullReload()
    // }
  })

  // TODO: 删除、添加
  watcher.on('change', (file) => {
    const path = slash(file)

    const isStoreDir = path.startsWith(`${options.storeDir}/`)

    /* const hotEvent: any = {
      // type: state、full-reload、hot-update
      // type: 'hot-update',
    } */

    if (isStoreDir) {
      const fileName = getPathName(path)
      // 如果修改的文件名: mutations、actions、getters
      if (HMR_MODULE_NAMES.includes(fileName)) {
        //
        server.ws.send({
          type: 'custom',
          event: 'vite-plugin-store-update',
          data: { // hotEvent
            type: 'hot-update',
          },
        })
      }
      else {
        fullReload()
      }
      /* for (const index in generateRoot.moduleOptions) {
        const moduleOption = generateRoot.moduleOptions[index]
        if (path === moduleOption.fullPath) {
          hotEvent = moduleOption
          break
        }
      } // end for
      // TODO: 防抖动
      if (['index', 'module'].includes(hotEvent.moduleInType)) {
        // 设置 type
        hotEvent.type = 'state'
      }
      else {
        hotEvent.type = 'hot-update'
      }
      server.ws.send({
        type: 'custom',
        event: 'vite-plugin-store-update',
        data: hotEvent,
      }) */
    }// end if(isStoreDir)
  })
}
