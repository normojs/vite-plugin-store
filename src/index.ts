import { resolve, basename } from 'path'
import type { Plugin, ResolvedConfig, ModuleNode } from 'vite'
import { Route, ResolvedOptions, UserOptions, PageDirOptions } from './types'
import { getFilesFromPath } from './files'
import { generateRoutes, generateClientCode, updateRouteFromHMR } from './generate'
import { debug, normalizePath } from './utils'
import { parseVueRequest } from './query'
import { resolveOptions } from './options'

const MODULE_ID = 'vite-plugin-store'

function storePlugin(userOptions: UserOptions = {}): Plugin {
  let config: ResolvedConfig | undefined
  let filesPath: string[] = []
  let generatedRoutes: Route[] | null | undefined
  let pagesDirPaths: string[] = []

  const options: ResolvedOptions = resolveOptions(userOptions)

  return {
    name: MODULE_ID,
    enforce: 'pre',
    configResolved(_config) {
      config = _config
      options.root = config.root
    },
    resolveId(id) {
      return MODULE_ID === id ? MODULE_ID : null
    },
    async load(id) {
      if (id === MODULE_ID) {
        if (!generatedRoutes) {
          generatedRoutes = []
          filesPath = []
          pagesDirPaths = []
          for (const pageDir of options.pagesDirOptions) {
            const pageDirPath = normalizePath(resolve(options.root, pageDir.dir))
            pagesDirPaths = pagesDirPaths.concat(pageDirPath)
            debug.gen('dir: %O', pageDirPath)

            const files = await getFilesFromPath(pageDirPath, options)
            filesPath = filesPath.concat(files.map(f => `${pageDirPath}/${f}`))
            debug.gen('files: %O', files)

            const routes = generateRoutes(files, pageDir, options)
            generatedRoutes = generatedRoutes.concat(routes)
          }
        }
        debug.gen('routes: %O', generatedRoutes)

        const clientCode = generateClientCode(generatedRoutes, options)
        // debug.gen('client code: %O', clientCode)

        return clientCode
      }
    },
    async transform(code: string, id: string) {
      const { query } = parseVueRequest(id)

      if (query && query.vue && query.type === 'route') {
        return {
          code: 'export default {}',
          map: null,
        }
      }
      return {
        code,
        map: null,
      }
    },
    generateBundle(_options, bundle) {
      if (options.replaceSquareBrackets) {
        const files = Object.keys(bundle).map(i => basename(i))
        for (const name in bundle) {
          const chunk = bundle[name]
          chunk.fileName = chunk.fileName.replace(/(\[|\])/g, '_')
          if (chunk.type === 'chunk') {
            for (const file of files)
              chunk.code = chunk.code.replace(file, file.replace(/(\[|\])/g, '_'))
          }
        }
      }
    },
    async handleHotUpdate({ file, server, read }) {
      const isPagesDir = pagesDirPaths.find(p => file.startsWith(`${p}/`))
      // Handle pages HMR
      if (isPagesDir && options.extensionsRE.test(file)) {
        let needReload = false

        // Handle new file
        if (!filesPath.includes(file)) {
          generatedRoutes = null
          needReload = true
        }
        else if (generatedRoutes) {
          const content = await read()
          needReload = updateRouteFromHMR(content, file, generatedRoutes, options)
        }

        if (needReload) {
          const { moduleGraph } = server
          const module = moduleGraph.getModuleById(MODULE_ID)
          if (module)
            server.moduleGraph.invalidateModule(module)

          debug.hmr('hmr update: %s', file.replace(options.root, ''))

          return [module!]
        }
      }
    },
  }
}

export * from './types'
export default storePlugin
