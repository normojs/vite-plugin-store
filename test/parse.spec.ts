import fs from 'fs'
import { resolve } from 'path'
import { parseSFC, parseCustomBlock } from '../src/parseSfc'
import { resolveOptions } from '../src/options'

const options = resolveOptions({})
const path = resolve('./test/assets/pages/blog/[id].vue')
const VueFile = fs.readFileSync(path, 'utf-8')

const expectCustomBlock = {
  meta: {
    requiresAuth: false,
  },
  name: 'name-override',
}

describe('Parser', () => {
  const parsed = parseSFC(VueFile)
  const customBlock = parsed.customBlocks.find(b => b.type === 'route')!
  test('SFC', () => {
    expect(parsed).toMatchSnapshot()
  })
  test('custom block', () => {
    const parsedCustomBlock = parseCustomBlock(customBlock, path, options)
    expect(parsedCustomBlock).toStrictEqual(expectCustomBlock)
  })
})
