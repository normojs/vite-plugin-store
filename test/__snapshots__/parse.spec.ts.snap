// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Parser SFC 1`] = `
Object {
  "cssVars": Array [],
  "customBlocks": Array [
    Object {
      "attrs": Object {},
      "content": "                        
                                 

             
             
  

         

          
                         
           

       
{
  meta: {
    requiresAuth: false,
  },
  name: 'name-override',
}
",
      "loc": Object {
        "end": Object {
          "column": 1,
          "line": 21,
          "offset": 230,
        },
        "source": "
{
  meta: {
    requiresAuth: false,
  },
  name: 'name-override',
}
",
        "start": Object {
          "column": 8,
          "line": 14,
          "offset": 160,
        },
      },
      "map": Object {
        "file": "anonymous.vue",
        "mappings": ";;;;;;;;;;;;;;AAcA;EACE,CAAC,CAAC,CAAC,CAAC,EAAE;IACJ,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,EAAE,CAAC,CAAC,CAAC,CAAC,CAAC;EACrB,CAAC;EACD,CAAC,CAAC,CAAC,CAAC,EAAE,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC;AACvB",
        "names": Array [],
        "sourceRoot": "",
        "sources": Array [
          "anonymous.vue",
        ],
        "sourcesContent": Array [
          "<script setup lang=\\"ts\\">
import { defineProps } from 'vue'

defineProps({
  id: String,
})

</script>

<template>
  blog/[id].vue: {{ id }}
</template>

<route>
{
  meta: {
    requiresAuth: false,
  },
  name: 'name-override',
}
</route>
",
        ],
        "version": 3,
      },
      "type": "route",
    },
  ],
  "filename": "anonymous.vue",
  "script": null,
  "scriptSetup": Object {
    "attrs": Object {
      "lang": "ts",
      "setup": true,
    },
    "content": "                        
import { defineProps } from 'vue'

defineProps({
  id: String,
})

",
    "lang": "ts",
    "loc": Object {
      "end": Object {
        "column": 1,
        "line": 8,
        "offset": 92,
      },
      "source": "
import { defineProps } from 'vue'

defineProps({
  id: String,
})

",
      "start": Object {
        "column": 25,
        "line": 1,
        "offset": 24,
      },
    },
    "setup": true,
    "type": "script",
  },
  "slotted": false,
  "source": "<script setup lang=\\"ts\\">
import { defineProps } from 'vue'

defineProps({
  id: String,
})

</script>

<template>
  blog/[id].vue: {{ id }}
</template>

<route>
{
  meta: {
    requiresAuth: false,
  },
  name: 'name-override',
}
</route>
",
  "styles": Array [],
  "template": Object {
    "ast": Object {
      "children": Array [
        Object {
          "content": "  blog/[id].vue: ",
          "loc": Object {
            "end": Object {
              "column": 18,
              "line": 11,
              "offset": 131,
            },
            "source": "
  blog/[id].vue: ",
            "start": Object {
              "column": 11,
              "line": 10,
              "offset": 113,
            },
          },
          "type": 2,
        },
        Object {
          "content": Object {
            "constType": 0,
            "content": "id",
            "isStatic": false,
            "loc": Object {
              "end": Object {
                "column": 23,
                "line": 11,
                "offset": 136,
              },
              "source": "id",
              "start": Object {
                "column": 21,
                "line": 11,
                "offset": 134,
              },
            },
            "type": 4,
          },
          "loc": Object {
            "end": Object {
              "column": 26,
              "line": 11,
              "offset": 139,
            },
            "source": "{{ id }}",
            "start": Object {
              "column": 18,
              "line": 11,
              "offset": 131,
            },
          },
          "type": 5,
        },
        Object {
          "content": "
",
          "loc": Object {
            "end": Object {
              "column": 1,
              "line": 12,
              "offset": 140,
            },
            "source": "
",
            "start": Object {
              "column": 26,
              "line": 11,
              "offset": 139,
            },
          },
          "type": 2,
        },
      ],
      "codegenNode": undefined,
      "isSelfClosing": false,
      "loc": Object {
        "end": Object {
          "column": 12,
          "line": 12,
          "offset": 151,
        },
        "source": "<template>
  blog/[id].vue: {{ id }}
</template>",
        "start": Object {
          "column": 1,
          "line": 10,
          "offset": 103,
        },
      },
      "ns": 0,
      "props": Array [],
      "tag": "template",
      "tagType": 0,
      "type": 1,
    },
    "attrs": Object {},
    "content": "
  blog/[id].vue: {{ id }}
",
    "loc": Object {
      "end": Object {
        "column": 1,
        "line": 12,
        "offset": 140,
      },
      "source": "
  blog/[id].vue: {{ id }}
",
      "start": Object {
        "column": 11,
        "line": 10,
        "offset": 113,
      },
    },
    "map": Object {
      "file": "anonymous.vue",
      "mappings": ";EAUE,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,EAAE,CAAC,EAAE,CAAC,EAAE,CAAC",
      "names": Array [],
      "sourceRoot": "",
      "sources": Array [
        "anonymous.vue",
      ],
      "sourcesContent": Array [
        "<script setup lang=\\"ts\\">
import { defineProps } from 'vue'

defineProps({
  id: String,
})

</script>

<template>
  blog/[id].vue: {{ id }}
</template>

<route>
{
  meta: {
    requiresAuth: false,
  },
  name: 'name-override',
}
</route>
",
      ],
      "version": 3,
    },
    "type": "template",
  },
}
`;
